app/display/representation-javafx/src/main/java/org/csstudio/display/builder/representation/javafx/widgets/ActionButtonRepresentation.java

```diff
@@ -139,7 +139,7 @@ private void checkModifiers(final MouseEvent event)
         }
 
         // 'control' ('command' on Mac OS X)
-        if (event.isShortcutDown())
+        if (event.isShortcutDown() || event.isMiddleButtonDown())

```

Comment:
Middle click already copies a PV name, same as it did in EDM. For actions buttons that write a PV, middle click gets that PV name.
Now it would open the display?

Reply:
If it's conflictual, I remove it.

Reply:
Ideally, you could get both:
Middle-click generally copies the PV name. Middle-click on text update, bool button, action button that writes PV, and you get the PV name. That pleases those who are used to fetching PV names that way from previous EDM usage.

Middle-click on action button that opens display, on the other hand, opens that in new tab, same as ctrl-click.
So you'd have to peek into the action: It is only one, is it the type that opens display -> do same as ctrl-click. Otherwise ignore the middle click, because it's already fetching a PV name.

Maybe you get that for free, because action with PV doesn't open anything, while action that opens display doesn't have a PV?

Reply:
In the code, middle click only modify the target of the OpenDisplayActionInfo, so it only affect a action button with the open display action. But when testing the version with the middle click, you can middle click the button with a write PV action and it does the same as a left click (it execute the action), so it could be conflictual (in the version without, middle click doesn't execute the action).

Do you want me to put a condition like "if (action instanceof OpenDisplayActionInfo)" before the middle click just to be sure ?

Reply:
Thanks for checking.
So middle-click to open display works as designed.
Copy of write-PV also still works.
Yes, if you can make middle-click on write-PV to _only_ get the PV name and not actually write, that'd be great.

Reply:
I added a verification before the middle click, so now, it behave like before if the button is not a open display only.

While testing, I noticed (version without the changes) that when you have multiple actions on the same button, it behave differently, regardless of the shortcuts. You can left / right click or middle click, it will execute the action selected (without considering shortcuts). I let that behavior in the version with my changes. So you can middle click to open displays in new tab on button with open display action only.

Reply:
sounds good, thanks